### ISSUE TITLE
ChatGPT API Key

### ISSUE THREAD

**Issue by DevNataneto** [lokinmodar/Echoglossian#112]
Olá! Gostaria de saber se só é possível usar a API do chatgpt no plugin com o GPT pago. 

Pois existe uma cota gratuita que pode ser usada na API gratuita, e queria saber se é possível utilizar.

Pergunto, pois tento copiar a chave da API e colar no campo do plugin e não cola

**Comment by lokinmodar** [comment]
Deveria ser possível sim, Não tem restrição nesse sentido... você deveria poder colar lá no campo sem problemas. 

> Olá! Gostaria de saber se só é possível usar a API do chatgpt no plugin com o GPT pago.
> 
> Pois existe uma cota gratuita que pode ser usada na API gratuita, e queria saber se é possível utilizar.
> 
> Pergunto, pois tento copiar a chave da API e colar no campo do plugin e não cola

**Comment by DevNataneto** [comment]
Quando eu dou ctrl + v ele não cola nada, como se a chave fosse grande demais(fiz o teste copiando apenas uma parte da chave e vai normalmente, mas a chave completa não)

**Comment by escojoncio** [comment]
By editing the json file from config i was able to put the entire key... But it does not work, or at least not for me, it gives me error 429 (quota exceeded), even if i have money on the account, so if anyone knows how to fix that...

edit: Seems like it cant manage that many request as it gives you that error when you log in, and after that if you look to the missions or things like that, it takes a HUGE load of time

So: Do not use chatGPT, it's not worth, save your money, thanks for the work of trying to add it!

**Comment by lokinmodar** [comment]
I'm checking the Plugin config window input size to see if there is a character limit (shouldn't have a limit but i'll review the code). As for the 429 error, Try checking if the API key you've generated is tied to the project that has the payment information setup in  the OpeAI API page

**Comment by escojoncio** [comment]
> I'm checking the Plugin config window input size to see if there is a character limit (shouldn't have a limit but i'll review the code). As for the 429 error, Try checking if the API key you've generated is tied to the project that has the payment information setup in the OpeAI API page

Just edited my post

**Comment by lokinmodar** [comment]
> > I'm checking the Plugin config window input size to see if there is a character limit (shouldn't have a limit but i'll review the code). As for the 429 error, Try checking if the API key you've generated is tied to the project that has the payment information setup in the OpeAI API page
> 
> Just edited my post

Oh i see. It seems their rate-limit sucks then :(

**Comment by escojoncio** [comment]
> > > I'm checking the Plugin config window input size to see if there is a character limit (shouldn't have a limit but i'll review the code). As for the 429 error, Try checking if the API key you've generated is tied to the project that has the payment information setup in the OpeAI API page
> > 
> > 
> > Just edited my post
> 
> Oh i see. It seems their rate-limit sucks then :(

Yeah... And it lags like hell, i mean, the game even freezes, thank you anyway for the hard work, 

Will be waiting for the new additions, hopefully someday we can change the game to french and still see the hud translated to our language, so we can enjoy the story without alterations (like english does) 

Thanks for all!

**Comment by lokinmodar** [comment]
The API Key bug is fixed as of v3.15.x but the rate-limit issue is under investigation

**Comment by lokinmodar** [comment]
No solution for this on our side. The rate-limit happens when the user is without credits on OpenAI
